Dockerfile

```diff
@@ -25,15 +22,13 @@ RUN apk add --no-cache libgcc libstdc++ build-base linux-headers
 # Set environment variables
 ENV PYTHONDONTWRITEBYTECODE=1
 ENV PYTHONUNBUFFERED=1
-ENV POETRY_NO_INTERACTION=1 \
-    POETRY_VIRTUALENVS_IN_PROJECT=1 \
-    POETRY_VIRTUALENVS_CREATE=1 \
-    POETRY_CACHE_DIR=/tmp/poetry_cache
+ENV UV_PROJECT_ENVIRONMENT=/app/.venv \
+    UV_CACHE_DIR=/tmp/uv_cache
 
 # Install python dependencies to a virtualenv
-COPY pyproject.toml poetry.lock ./
-RUN pip install poetry=="${poetry_version}"
-RUN poetry install --without dev --no-root && rm -rf $POETRY_CACHE_DIR
+COPY pyproject.toml uv.lock ./
+RUN pip install uv==0.9.5
+RUN uv sync --frozen --no-dev && rm -rf $UV_CACHE_DIR
```

Comment:
can we add `--compile-bytecode` here? https://docs.astral.sh/uv/guides/integration/docker/#compiling-bytecode

we might also need `--no-editable` although I'm not sure

> In the context of a multi-stage Docker image, --no-editable can be used to include the project in the synced virtual environment from one stage, then copy the virtual environment alone (and not the source code) into the final image.